### ISSUE TITLE
Delta Lake connector

### ISSUE THREAD

**Issue by findepi** [trinodb/trino#11296]
- [ ] https://github.com/trinodb/trino/pull/10897
- [ ]

**Comment by findepi** [comment]
cc @jirassimok @alexjo2144

**Comment by homar** [comment]
Based on TODOs in code I created following issues related to Delta Lake connector:
https://github.com/trinodb/trino/issues/12004
https://github.com/trinodb/trino/issues/12005
https://github.com/trinodb/trino/issues/12007
https://github.com/trinodb/trino/issues/12008
https://github.com/trinodb/trino/issues/12009
https://github.com/trinodb/trino/issues/12010
https://github.com/trinodb/trino/issues/12011
https://github.com/trinodb/trino/issues/12012
https://github.com/trinodb/trino/issues/12013
https://github.com/trinodb/trino/issues/12014
https://github.com/trinodb/trino/issues/12018
https://github.com/trinodb/trino/issues/12028
https://github.com/trinodb/trino/issues/12029
https://github.com/trinodb/trino/issues/12030
https://github.com/trinodb/trino/issues/12031
https://github.com/trinodb/trino/issues/12032
https://github.com/trinodb/trino/issues/12033
https://github.com/trinodb/trino/issues/12034
https://github.com/trinodb/trino/issues/12040
https://github.com/trinodb/trino/issues/12041

**Comment by findepi** [comment]
@homar thanks!
 i moved the above list into issue description. Feel free to remove checkboxes from your comment (or the list)

**Comment by Trymzet** [comment]
Do I get it right in that vanilla Databricks is not yet supported? This connector requires using the Thrift schema for Hive connection string, and AFAIK Databricks only exposes the JDBC connection string for its Hive metastore (eg `jdbc:spark://adb-123456789.5.azuredatabricks.net:443/default;transportMode=http;ssl=1;httpPath=sql/protocolv1/o/123456789/0427-122644-45iadnd;AuthMech=3;UID=token;PWD=<personal-access-token>`). You can only use Thrift if you set up a custom metastore for Databricks.

**Comment by findepi** [comment]
> You can only use Thrift if you set up a custom metastore for Databricks.

Yes.
Or, use Glue.

> AFAIK Databricks only exposes the JDBC connection string (eg `jdbc:spark://adb-123456789.5.azuredatabricks.net:443/default;transportMode=http;ssl=1;httpPath=sql/protocolv1/o/123456789/0427-122644-45iadnd;AuthMech=3;UID=token;PWD=<personal-access-token>`).

We have no plans to connect to Databricks runtime using Databricks JDBC.
This would kill most benefits of this connector.

**Comment by alexjo2144** [comment]
Right. here's the Databricks docs for setting up an external HMS or Glue https://docs.databricks.com/data/metastores/index.html

**Comment by Trymzet** [comment]
@alexjo2144 Thanks. I was researching if we can use dbt with various sources all through Trino (inspired by this [video](https://trino.io/episodes/21.html)) and it seems that Databricks is doable as well, although integrating directly through `dbt-databricks` plugin is more straightforward. For future generaitons: using Databricks through `dbt-trino` plugin requires setting up and maintaining your own Hive instance and creating a global [init script](https://docs.databricks.com/clusters/init-scripts.html) to set the config of each cluster to use that Hive. Also, DBFS is [not supported](https://docs.databricks.com/data/metastores/external-hive-metastore.html#hive-metastore-deployment-modes) with this method.

**Comment by PragyaJaiswal** [comment]
I would think this should change things: https://www.databricks.com/blog/extending-databricks-unity-catalog-open-apache-hive-metastore-api. Does that look promising?

**Comment by findepi** [comment]
Trino now has Delta Lake connector for some time now. Of course there are follow-up issues, as always, and these are tracked separately.
This roadmap issue is not being kept up to date as a way to organize future work on the feature.

Good job, everyone involved. 👏
